autoblocks/_impl/tracer.py

```diff
@@ -101,6 +123,70 @@ def start_span(self):
                 props["span_id"] = prev_span_id
             self.set_properties(props)
 
+    async def _evaluate_event(self, event: TracerEvent, evaluator: BaseEventEvaluator) -> Optional[Evaluation]:
+        """
+        Evaluates an event using a provided evaluator.
+        """
+        evaluation = None
+        if inspect.iscoroutinefunction(evaluator.evaluate_event):
+            try:
+                evaluation = await evaluator.evaluate_event(event=event)
+            except Exception as err:
+                return Evaluation(
+                    score=0,
+                    metadata={
+                        "error": "Evaluation raised an exception during execution",
+                        "errorDetail": f"{repr(err)}",
+                    },
+                )
+        else:
+            try:
+                ctx = contextvars.copy_context()
+                evaluation = await global_state.event_loop().run_in_executor(
+                    None,
+                    ctx.run,
+                    evaluator.evaluate_event,
+                    event,
+                )
+            except Exception as err:
+                return Evaluation(
+                    score=0,
+                    metadata={
+                        "error": "Evaluation raised an exception during execution",
+                        "errorDetail": f"{repr(err)}",
+                    },
+                )
+
+        return evaluation
+
+    async def _run_evaluators(
+        self, evaluators: List[BaseEventEvaluator], event: TracerEvent, max_evaluator_concurrency: int
+    ) -> List[ClickHouseEvaluationObject]:
+        if len(evaluators) == 0:
+            return []
+        try:
+            evaluations: List[Evaluation] = await gather_with_max_concurrency(
```

Comment:
just fyi i verified `asyncio.gather` returns stuff in the same order given:

https://docs.python.org/3/library/asyncio-task.html#asyncio.gather

> If all awaitables are completed successfully, the result is an aggregate list of returned values. The order of result values corresponds to the order of awaitables in aws.